third_party/com_github_aws_cpp_sdk/workspace.bzl

```diff
@@ -0,0 +1,38 @@
+# Copyright 2024 The TensorStore Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+load("//third_party:repo.bzl", "third_party_http_archive")
+load("@bazel_tools//tools/build_defs/repo:utils.bzl", "maybe")
+
+def repo():
+    maybe(
+        third_party_http_archive,
+        name = "com_github_aws_cpp_sdk",
+        patch_cmds = [
```

Comment:
Nice patch.  Have you tried this with the cmake build?

Reply:
I copied it from here https://github.com/tensorflow/io/blob/4d70341df05ddce09c257d4ecaaaba1873594105/WORKSPACE#L224-L236

Yeah this falls over on `cmake -S . -B build` with the following truncated output:

```bash
- verifying file...
       file='/home/simon/code/tensorstore/build/_deps/aws_cpp_sdk-subbuild/aws_cpp_sdk-populate-prefix/src/1.8.187.tar.gz'
-- Downloading... done
-- extracting...
     src='/home/simon/code/tensorstore/build/_deps/aws_cpp_sdk-subbuild/aws_cpp_sdk-populate-prefix/src/1.8.187.tar.gz'
     dst='/home/simon/code/tensorstore/build/_deps/aws_cpp_sdk-src'
-- extracting... [tar xfz]
-- extracting... [analysis]
-- extracting... [rename]
-- extracting... [clean up]
-- extracting... done
[ 33%] No update step for 'aws_cpp_sdk-populate'
[ 44%] Performing patch step for 'aws_cpp_sdk-populate'
sed: -e expression #1, char 1: unknown command: `''
gmake[2]: *** [CMakeFiles/aws_cpp_sdk-populate.dir/build.make:118: aws_cpp_sdk-populate-prefix/src/aws_cpp_sdk-populate-stamp/aws_cpp_sdk-populate-patch] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:83: CMakeFiles/aws_cpp_sdk-populate.dir/all] Error 2
gmake: *** [Makefile:91: all] Error 2

CMake Error at /usr/share/cmake-3.29/Modules/FetchContent.cmake:1714 (message):
  Build step for aws_cpp_sdk failed: 2
Call Stack (most recent call first):
  /usr/share/cmake-3.29/Modules/FetchContent.cmake:1854:EVAL:2 (__FetchContent_directPopulate)
  /usr/share/cmake-3.29/Modules/FetchContent.cmake:1854 (cmake_language)
  /usr/share/cmake-3.29/Modules/FetchContent.cmake:2081 (FetchContent_Populate)
  build/third_party/CMakeLists.txt:3 (FetchContent_MakeAvailable)
```

The contents of the patch file (aws_cpp_sdk-populate-patch-info.txt) look like extraneous `;` characters have been added in place of spaces in the `patch_cmds`.

```bash
~/code/tensorstore/build/_deps/aws_cpp_sdk-subbuild/aws_cpp_sdk-populate-prefix/src/aws_cpp_sdk-populate-stamp$ cat aws_cpp_sdk-populate-patch-info.txt 
# This is a generated file and its contents are an internal implementation detail.
# The update step will be re-executed if anything in this file changes.
# No other meaning or use of this file is supported.

command=sed;-i.bak;'s/UUID::RandomUUID/Aws::Utils::UUID::RandomUUID/g';aws-cpp-sdk-core/source/client/AWSClient.cpp;&&;sed;-i.bak;'s/__attribute__;(;(;visibility;(;default;);););//g';aws-cpp-sdk-core/include/aws/core/external/tinyxml2/tinyxml2.h;&&;/usr/bin/cmake;-E;copy;/home/simon/code/tensorstore/third_party/com_github_aws_cpp_sdk/aws_cpp_sdk.BUILD.bazel;BUILD.bazel;&&;/usr/bin/cmake;-E;copy;/home/simon/code/tensorstore/build/third_party/aws_cpp_sdk-proxy-CMakeLists.txt;CMakeLists.txt
work_dir=/home/simon/code/tensorstore/build/_deps/aws_cpp_sdk-src
```

Any insights here? Maybe `http_archive` does some string handling that `third_party_http_archive` does not yet?

Reply:
`patch_cmds` was converted into a `.diff file` and applied with `patch`.